plugins/jobs/src/js/components/JobsForm.tsx

```diff
@@ -0,0 +1,214 @@
+import { Trans } from "@lingui/macro";
+import { i18nMark, withI18n } from "@lingui/react";
+import classNames from "classnames";
+import React, { Component } from "react";
+
+import ErrorsAlert from "#SRC/js/components/ErrorsAlert";
+import FluidGeminiScrollbar from "#SRC/js/components/FluidGeminiScrollbar";
+import JSONEditor from "#SRC/js/components/JSONEditor";
+import TabButton from "#SRC/js/components/TabButton";
+import TabButtonList from "#SRC/js/components/TabButtonList";
+import Tabs from "#SRC/js/components/Tabs";
+import TabView from "#SRC/js/components/TabView";
+import TabViewList from "#SRC/js/components/TabViewList";
+
+import "brace/mode/json";
+import "brace/theme/monokai";
+import "brace/ext/language_tools";
+
+import {
+  FormError,
+  JobFormActionType,
+  Action,
+  JobSpec,
+  JobOutput
+} from "./form/helpers/JobFormData";
+import GeneralFormSection from "./form/GeneralFormSection";
+import {
+  jobSpecToOutputParser,
+  jobSpecToFormOutputParser
+} from "./form/helpers/JobParsers";
+
+const ServiceErrorPathMapping: any[] = [];
+
+interface JobFormProps {
+  onChange: (action: Action) => void;
+  jobSpec: JobSpec;
+  activeTab: string;
+  handleTabChange: (tab: string) => void;
+  errors: any[];
+  isJSONModeActive: boolean;
+  onErrorsChange: (errors: FormError[], type?: string) => void;
+  showAllErrors: boolean;
+}
+
+interface NavigationItem {
+  id: string;
+  key: string;
+  label: string;
+  className?: string;
+  children?: NavigationItem[];
+}
+
+class JobModalForm extends Component<JobFormProps, {}> {
+  constructor(props) {
+    super(props);
+
+    this.onInputChange = this.onInputChange.bind(this);
+    this.handleJSONChange = this.handleJSONChange.bind(this);
+    this.handleJSONErrorStateChange = this.handleJSONErrorStateChange.bind(
+      this
+    );
+  }
+
+  onInputChange(event: any) {
```

Comment:
Let's use the proper type for the event here https://github.com/sw-yx/react-typescript-cheatsheet#forms-and-events.